### ISSUE TITLE
BoxResize box assignment updates initial reference

### ISSUE THREAD

**Issue by joaander** [glotzerlab/hoomd-blue#705]
## Description

BoxResize holds onto the initial BoxDim it is given by reference. When assigning a new Python Box object, the original reference is updated by the Box assignment operator.

## Script

```python
# Include a minimal script that reproduces the problem
import hoomd

device = hoomd.device.CPU()
sim = hoomd.Simulation(device)
sim.create_state_from_gsd('test.gsd')

box1_initial = hoomd.Box.cube(10)
box2_initial = hoomd.Box.cube(20)
resize = hoomd.update.BoxResize(box1_initial, box2_initial, hoomd.variant.Ramp(0, 0, 1, 100000), 100)

sim.operations.updaters.append(resize)
sim.run(1)
print("box1_initial:", box1_initial.Lx, " / resize.box1: ", resize.box1.Lx)
print(resize.box1 is box1_initial)

box1_initial.Lx = 25
print("box1_initial:", box1_initial.Lx, " / resize.box1: ", resize.box1.Lx)
print(resize.box1 is box1_initial)

box1_update = hoomd.Box.cube(30)
resize.box1 = box1_update
print("box1_initial:", box1_initial.Lx, " / box1_update: ", box1_update.Lx, " / resize.box1: ", resize.box1.Lx)
print(resize.box1 is box1_update)

```

## Output

```
box1_initial: 10.0  / resize.box1:  10.0
True
box1_initial: 25.0  / resize.box1:  25.0
True
box1_initial: 30.0  / box1_update:  30.0  / resize.box1:  30.0
True
```

## Expected output

```
box1_initial: 10.0  / resize.box1:  10.0
True
box1_initial: 25.0  / resize.box1:  25.0
True
box1_initial: 25.0  / box1_update:  30.0  / resize.box1:  30.0
True
```

## Other notes

We could quickly fix this bug by using pointers instead of references in `BoxResize`, however the existence of the bug exposes an anti-pattern - two member variables that need to be kept in sync. Perhaps we could solve this like we do with Trigger and ParticleFilter and make python features of `Box` available as a subclass of `BoxDim`?

**Comment by joaander** [comment]
While this bug needs to be fixed, it is not critical for the 3.0.0 release.

**Comment by b-butler** [comment]
I recommend `Box` inheriting from `BoxDim`. I don't know why I didn't do that to start.
